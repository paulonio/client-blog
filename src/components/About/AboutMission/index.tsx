import React from 'react';
import Mission from '@/components/AboutUs/Mission';

import styles from './styled.module.scss';

const AboutMission = () => {
  return (
    <section className={styles.missions}>
      <Mission />
      <Mission />
    </section>
  );
};

export default AboutMission;
