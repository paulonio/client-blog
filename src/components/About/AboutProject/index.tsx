import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import { AboutContentType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface AboutProjectProps {
  content: AboutContentType;
}

const AboutProject: FC<AboutProjectProps> = ({ content }) => {
  const { urlToImage } = content;
  const { t } = useTranslation('about');

  return (
    <section className={styles.wrapper}>
      <div>
        <img src={urlToImage} alt="About project" />
      </div>
      <div className={styles.content__wrapper}>
        <h2 className={sen.className}>{t('project.title')}</h2>
        <h4 className={sen.className}>{t('project.subtitle')}</h4>
        <p className="body-s">{t('project.body')}</p>
      </div>
    </section>
  );
};

export default AboutProject;
