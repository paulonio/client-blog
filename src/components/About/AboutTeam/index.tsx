import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import { AboutContentType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface AboutTeamProps {
  content: AboutContentType;
}

const AboutTeam: FC<AboutTeamProps> = ({ content }) => {
  const { urlToImage } = content;
  const { t } = useTranslation('about');

  return (
    <section className={styles.wrapper}>
      <div className={styles.content__wrapper}>
        <h2 className={sen.className}>{t('team.title')}</h2>
        <h4 className={sen.className}>{t('team.subtitle')}</h4>
        <p className="body-s">{t('team.body')}</p>
      </div>
      <div>
        <img src={urlToImage} alt="About team" />
      </div>
    </section>
  );
};

export default AboutTeam;
