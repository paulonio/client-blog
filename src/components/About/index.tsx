import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import { InfoType } from '@/types/types';

import styles from './styled.module.scss';
import { sen } from '@/styles/fonts';
import { stringToKey } from '@/utils/utils';

interface AboutProps {
  info: InfoType[];
}

const About: FC<AboutProps> = ({ info }) => {
  const { t } = useTranslation('about');

  return (
    <section className={styles.about}>
      <div className={styles.heading}>
        <div className={styles.about__title}>
          <h5>{t('header')}</h5>
          <h1>{t('title')}</h1>
        </div>
        <p className={`${styles.about__description} body-s`}>
          {t('body')} {t('second-part')}
        </p>
      </div>
      <div>
        <div className={styles.info}>
          {info.map(({ amount, description }) => (
            <div key={description} className={styles.info__item}>
              <p className={`${sen.className} display`}>{amount}</p>
              <p className="body-l">{t(stringToKey(description))}</p>
            </div>
          ))}
        </div>
        <div className={styles.patterns__wrapper}>
          <div className={styles.pattern__left}>
            <div />
            <div />
          </div>
          <div className={styles.pattern__right} />
        </div>
      </div>
    </section>
  );
};

export default About;
