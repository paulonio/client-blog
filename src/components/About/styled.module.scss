@import '../../styles/sass/constants';

.about {
  padding-top: 80px;
  padding-left: 109px;
  display: flex;
  flex-direction: column;
  row-gap: 202px;
  background: url('/assets/images/special.jpg');
  background-position: bottom;
  background-repeat: no-repeat;

  @media screen and (max-width: 768px) {
    text-align: center;
    padding: 10px 32px;
  }

  @media screen and (max-width: 555px) {
    padding: 8px 10px;
  }

  @media screen and (max-width: 400px) {
    padding: 8px 0;
  }

  &__title {
    width: 100%;
    max-width: 630px;
    height: 100%;
    padding: 64px;
    background-color: $color-white;

    @media screen and (max-width: 768px) {
      padding: 0;
    }

    & h1 {
      margin-top: 16px;
      color: $color-black;
    }
  }
}

.heading {
  display: flex;
  
  @media screen and (max-width: 768px) {
    flex-direction: column;
  }
  
  & p {
    width: 100%;
    max-width: 541px;
    padding: 100px 32px 79px;

    @media screen and (max-width: 768px) {
      max-width: 768px;
      background-color: $color-white;
      padding: 10px;
    }
  }
}

.info {
  width: 100%;
  max-width: 630px;
  padding: 32px 48px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: $color-primary;

  @media screen and (max-width: 555px) {
    padding: 8px 10px;
  }
}

.patterns__wrapper {
  display: flex;
}

.pattern {
  &__left {
    width: 100%;
    max-width: 630px;
    display: flex;
    height: 23px;
    background-color: $color-primary;
  
    & div {
      width: 50%;
  
      &:first-child {
        background-color: $color-secondary;
      }
    }
  }

  &__right {
    width: 100%;
    max-width: 232px;
  
    height: 23px;
    background-color: $color-primary;
  }
}
