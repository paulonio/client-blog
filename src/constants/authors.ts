import { Author } from '@/types/types';

export const AUTHORS: Author[] = [
  {
    fullName: 'Floyd Miles',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Floyd.jpg',
  },
  {
    fullName: 'Dianne Russell',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Dianne.jpg',
  },
  {
    fullName: 'Jenny Wilson',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Jenny.jpg',
  },
  {
    fullName: 'Leslie Alexander',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Leslie.jpg',
  },
];
