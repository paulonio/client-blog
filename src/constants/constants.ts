import BusinessIcon from '@/icons/BusinessIcon';
import CyborgIcon from '@/icons/CyborgIcon';
import EconomyIcon from '@/icons/EconomyIcon';
import ShuttleIcon from '@/icons/ShuttleIcon';

import { About, Category } from '@/types/types';

export const ABOUT_US: About = {
  title: 'We are a community of content writers who share their learnings',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
};

export const OUR_MISSION: About = {
  title: 'Creating valuable content for creatives all around the world',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
};

export const CATEGORY_ITEMS: Category[] = [
  {
    title: 'Business',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: BusinessIcon,
  },
  {
    title: 'Startup',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: ShuttleIcon,
  },
  {
    title: 'Economy',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: EconomyIcon,
  },
  {
    title: 'Technology',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: CyborgIcon,
  },
];

export const SPECIAL_POST: About = {
  title: 'It started out as a simple idea and evolved into our passion',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
};
