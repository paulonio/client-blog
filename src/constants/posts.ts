import type { PostType } from '@/types/types';

export const POSTS: PostType[] = [
  {
    urlToAvatar: '/assets/images/Dianne.jpg',
    username: 'John Doe',
    date: 'May 23, 2022',
    tag: 'BUSINESS',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/featured-post.jpg',
  },
  {
    urlToAvatar: '/assets/images/Floyd.jpg',
    username: 'John Doe',
    date: 'Aug 23, 2022',
    tag: 'EXPERIENCE',
    title: 'Step-by-step guide to choosing great font pairs',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/blog-featured.jpg',
  },
  {
    urlToAvatar: '/assets/images/Jenny.jpg',
    username: 'John Doe',
    date: 'Sep 23, 2022',
    tag: 'SCREEN',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/design-post.jpg',
  },
  {
    urlToAvatar: '/assets/images/Jonathan.jpg',
    username: 'John Doe',
    date: 'May 23, 2022',
    tag: 'TECHNOLOGY',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/build-rapport.jpg',
  },
  {
    urlToAvatar: '/assets/images/Leslie.jpg',
    username: 'John Doe',
    date: 'Jan 23, 2022',
    tag: 'MARKETING',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/logo-design.jpg',
  },
  {
    urlToAvatar: '/assets/images/Jenny.jpg',
    username: 'John Doe',
    date: 'May 23, 2022',
    tag: 'LIFE',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/figma-design.jpg',
  },
  {
    urlToAvatar: '/assets/images/Dianne.jpg',
    username: 'John Doe',
    date: 'May 23, 2022',
    tag: 'BUSINESS',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/featured-post.jpg',
  },
  {
    urlToAvatar: '/assets/images/.jpg',
    username: 'John Doe',
    date: 'May 23, 2022',
    tag: 'EXPERIENCE',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.',
    body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.',
    urlToImage: '/assets/images/featured-post.jpg',
  },
];
