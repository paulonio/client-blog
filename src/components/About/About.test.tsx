import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import About from '@/components/About';
import { ABOUT_INFO } from '@/constants/about-us';
import type { InfoType } from '@/types/types';

const FAKE_DATA: InfoType[] = [{ amount: '100', description: 'test-items' }];

describe('About page', () => {
  it('should render about page', () => {
    const { container } = render(<About info={ABOUT_INFO} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with fake data', () => {
    render(<About info={FAKE_DATA} />);

    const fakeItem = screen.getByText('test-items');

    expect(fakeItem).toBeInTheDocument();
  });

  it('should render with empty array', () => {
    const { container } = render(<About info={[]} />);

    expect(container).toMatchSnapshot();
  });
});
