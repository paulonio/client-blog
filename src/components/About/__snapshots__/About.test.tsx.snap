// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page should render about page 1`] = `
<div>
  <section
    class="about"
  >
    <div
      class="heading"
    >
      <div
        class="about__title"
      >
        <h5>
          header
        </h5>
        <h1
          data-testid="about-title"
        >
          title
        </h1>
      </div>
      <p
        class="about__description body-s"
      >
        body
         
        second-part
      </p>
    </div>
    <div>
      <div
        class="info"
      >
        <div
          class="info__item"
        >
          <p
            class="className display"
          >
            12+
          </p>
          <p
            class="body-l"
          >
            blogs-published
          </p>
        </div>
        <div
          class="info__item"
        >
          <p
            class="className display"
          >
            18K+
          </p>
          <p
            class="body-l"
          >
            views-on
          </p>
        </div>
        <div
          class="info__item"
        >
          <p
            class="className display"
          >
            30K+
          </p>
          <p
            class="body-l"
          >
            total-active
          </p>
        </div>
      </div>
      <div
        class="patterns__wrapper"
      >
        <div
          class="pattern__left"
        >
          <div />
          <div />
        </div>
        <div
          class="pattern__right"
        />
      </div>
    </div>
  </section>
</div>
`;

exports[`About page should render with empty array 1`] = `
<div>
  <section
    class="about"
  >
    <div
      class="heading"
    >
      <div
        class="about__title"
      >
        <h5>
          header
        </h5>
        <h1
          data-testid="about-title"
        >
          title
        </h1>
      </div>
      <p
        class="about__description body-s"
      >
        body
         
        second-part
      </p>
    </div>
    <div>
      <div
        class="info"
      />
      <div
        class="patterns__wrapper"
      >
        <div
          class="pattern__left"
        >
          <div />
          <div />
        </div>
        <div
          class="pattern__right"
        />
      </div>
    </div>
  </section>
</div>
`;
